rbac/api/cross_access/view.py

```diff
@@ -211,3 +283,52 @@ def create_principal(self, target_account, user_id):
             cross_account_principal = Principal.objects.get_or_create(username=principal_name, cross_account=True)
 
         return cross_account_principal
+
+    def validate_and_get_input_for_update(self, request_data):
+        """Validate the update api input."""
+        target_account = request_data.get("target_account")
+        start_date = request_data.get("start_date")
+        end_date = request_data.get("end_date")
+        roles = request_data.get("roles")
+
+        try:
+            if start_date and isinstance(start_date, str):
+                start_date = pytz.utc.localize(datetime.strptime(start_date, "%m/%d/%Y"))
+            if end_date and isinstance(end_date, str):
+                end_date = pytz.utc.localize(datetime.strptime(end_date, "%m/%d/%Y"))
+        except ValueError:
+            raise self.throw_validation_error(
+                "cross-account-update", "start_date or end_date does not match format: '%m/%d/%Y'."
+            )
+
+        if start_date and start_date > pytz.utc.localize(datetime.now() + timedelta(60)):
+            raise self.throw_validation_error("cross-account-update", "Start date must be within 60 days of today.")
```

Comment:
I wonder...will we get into a situation where you could potentially perpetually update a request to bump it out 60 days? Probably an edge-case that may not be a concern, but I just thought about that.